ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -317,7 +318,9 @@ struct ggml_cuda_pool_vmm : public ggml_cuda_pool {
 
     ~ggml_cuda_pool_vmm() {
         if (pool_addr != 0) {
-            CU_CHECK(cuMemUnmap(pool_addr, pool_size));
+            for (std::pair<CUdeviceptr, size_t>& mapping : mappings) {
+                CU_CHECK(cuMemUnmap(mapping.first, mapping.second));
+            }
```

Comment:
Since this is only necessary for HIP, I would prefer if the current behavior is kept for CUDA.

Also please do not use C++ style casts and try to maintain the code formatting. You can use the included `.clang-format` if unsure about the style.